File: src/main/java/org/jukeboxmc/block/BlockStickypistonarmcollision.java
package org.jukeboxmc.block;

import org.jukeboxmc.block.direction.BlockFace;
import org.jukeboxmc.item.ItemStickypistonarmcollision;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockStickypistonarmcollision extends Block {

    public BlockStickypistonarmcollision() {
        super( "minecraft:sticky_piston_arm_collision" );
    }

    @Override
    public ItemStickypistonarmcollision toItem() {
        return new ItemStickypistonarmcollision();
    }

    @Override
    public BlockType getType() {
        return BlockType.STICKYPISTONARMCOLLISION;
    }

    public void setBlockFace( BlockFace blockFace ) {
        this.setState( "facing_direction", blockFace.ordinal() );
    }

    public BlockFace getBlockFace() {
        return this.stateExists( "facing_direction" ) ? BlockFace.values()[this.getIntState( "facing_direction" )] : BlockFace.NORTH;
    }

}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockStickyPistonArmCollision.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.StickyPistonArmCollision
import org.jukeboxmc.api.block.data.BlockFace
import org.jukeboxmc.server.block.JukeboxBlock

class BlockStickyPistonArmCollision(identifier: Identifier, blockStates: NbtMap?) :
    JukeboxBlock(identifier, blockStates), StickyPistonArmCollision {

    override fun getFacingDirection(): BlockFace {
        return BlockFace.entries[this.getIntState("facing_direction")]
    }

    override fun setFacingDirection(value: BlockFace): StickyPistonArmCollision {
        return this.setState("facing_direction", value.ordinal)
    }
}